src/App.jsx

```diff
@@ -10,18 +10,40 @@ export default function App() {
 
   const register = async (e) => {
     e.preventDefault();
-    // Write your register code here
-
+    try {
+      const response = await fetch('/register', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(user),
+      });
+      const data = await response.json();
+      setRegisterResponse(data.message);
+    } catch (error) {
+      console.error('Error registering:', error);
+      setRegisterResponse('Error registering. Please try again.');
+    }
   };
 
   const login = async (e) => {
     e.preventDefault();
-    // Write your login code here
-
+    try {
+      const response = await fetch('/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(user),
+      });
+      const data = await response.json();
```

Comment:
After Login you should save the token to local storage. 
Do that and you pass the exercise.